### PR TITLE
WP_Query::get_queried_object() before get_posts() for authors

### PR DESCRIPTION
**Pull request by dd32**
See https://core.trac.wordpress.org/ticket/55100

### CODE REVIEW

File: tests/phpunit/tests/query.php [hunk 1]
```diff
@@ -696,4 +696,32 @@ public function test_query_vars_should_match_first_of_multiple_terms() {
...
+		$post_id = self::factory()->post->create(
+			array(
+				'post_author' => $user_id,
+			)
+		);
+
+		$query = new WP_Query();
+		$query->parse_query(
+			array(
+				'author' => $user_id
```
Comment (swissspidy) [inline tests/phpunit/tests/query.php]:
```suggestion
				'author' => $user_id,
```

File: tests/phpunit/tests/query.php [hunk 2]
```diff
@@ -696,4 +696,32 @@ public function test_query_vars_should_match_first_of_multiple_terms() {
...
+			)
+		);
+		$this->assertEquals( $query->get_queried_object(), $user );
+
+		$query->parse_query(
+			array(
+				'author_name' => $user->user_nicename
```
Comment (swissspidy) [inline tests/phpunit/tests/query.php]:
```suggestion
				'author_name' => $user->user_nicename,
```

File: src/wp-includes/class-wp-query.php [hunk 3]
```diff
@@ -3614,6 +3614,12 @@ public function get_queried_object() {
 			$this->queried_object_id = (int) $this->post->ID;
 		} elseif ( $this->is_author ) {
 			$this->queried_object_id = (int) $this->get( 'author' );
+			if ( ! $this->queried_object_id ) {
+				$user = get_user_by( 'slug', $this->get( 'author_name' ) );
```
Comment (swissspidy) [inline src/wp-includes/class-wp-query.php:3618]: In your workaround PR you first checked whether `$this->get( 'author_name' )` is truthy. Worth doing that here as well?

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in https://core.trac.wordpress.org/changeset/52822.

File: src/wp-includes/class-wp-query.php (hunk 3 above)
Comment (SergeyBiryukov) [reply src/wp-includes/class-wp-query.php:3618]: Good point, I've updated the code in the commit to match how it's done for taxonomies.

**Pull request closed without merging**